### ISSUE TITLE
Flutter tooling silently ignores Android SDK path discrepancy

### ISSUE THREAD

**Issue by mravn-google** [flutter/flutter#13750]
See https://github.com/flutter/flutter/issues/13353 for context.

## Steps to Reproduce

1. Have two Android SDKs on your system, A and B. Assume A does not work with Flutter (maybe because it is too old). Assume `ANDROID_HOME` environment variable points to A, and you don't want to change that because you need A for other projects.
1. `flutter create xxx`, `cd xxx`, `flutter run`. It fails because A does not work with Flutter, by assumption.
1. Retarget Flutter to use B: `flutter config --android-sdk path_to_B`
1. `flutter run` still fails.
1. Observe `xxx/android/local.properties` still points to A.

We should allow `local.properties` to override the Flutter config setting on a project-by-project basis, but I think it would make sense to issue a warning when the two are in disagreement.

**Comment by db-007** [comment]
I have had same issue as reported in [https://github.com/flutter/flutter/issues/13353](https://github.com/flutter/flutter/issues/13353).  My ANDROID_HOME env var was set to point to another Android SDK location (one installed by Xamarin installation).
But my Android Studio Android SDK path setting (**File>Appearances & Behavior>SDK**) was always pointing to the Android SDK it installed, so I am not sure why flutter was picking up the other one, I guess it was looking in ANDROID_HOME env var and using that.
One note though, I did issue `flutter config --android-sdk path_to_my_correct_sdk` but that did not fix anything.  My **app/android/local.properties** was still pointing to the Xamarin Android SDK for some reason and as a result my app was still not installing and running after `flutter run`.
Only manually updating path in `local.properties` as you suggested fixed the issue.  I think there should be some way of detecting multiple Android SDK installations and asking user to pick one.  That, I assume may take time, so maybe even better to have clear instructions in documentation Getting Started section to set this path inside` local.properties` on project-by-project basis would be a better solution.

**Comment by jensjoha** [comment]
I ran into this last night.

My PATH was pointing to an old android sdk where "--android-licenses" didn't work. (I have no "ANDROID_HOME" env variable). I updated my path to point to the new android sdk, and `flutter doctor --android-licenses` now worked and I accepted my licences (i.e. flutter doctor used the new sdk path).

`flutter build apk` and `flutter run` still didn't work though, because (as it turned out) gradle uses the sdk path from android/local.properties.
Updating it manually fixed the problem, though a hint from flutter doctor would have saved me 1+ hours of diagnosing...

```
[...]

[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at <new path>
  [...]

>flutter build apk
Initializing gradle...                                0,8s
Resolving dependencies...                                \

* Error running Gradle:
Unable to download needed Android SDK components, as the following licenses have not been accepted:
Android SDK Build-Tools 26.0.2, Android SDK Platform 27

To resolve this, please run the following command in a Terminal:
flutter doctor --android-licenses

>flutter doctor --android-licenses
All SDK package licenses accepted.======] 100% Computing updates...


>flutter build apk
Initializing gradle...                                0,8s
Resolving dependencies...                                \

* Error running Gradle:
Unable to download needed Android SDK components, as the following licenses have not been accepted:
Android SDK Build-Tools 26.0.2, Android SDK Platform 27

To resolve this, please run the following command in a Terminal:
flutter doctor --android-licenses
```

**Comment by willianra** [comment]
paso 1 desinstalar java 
paso 2 descargar la ultima version de java   
paso 3 actualizar variables de entorno
paso 4 flutter doctor --android-licenses

**Comment by powext** [comment]
How is this still open....

**Comment by gnprice** [comment]
This issue no longer reproduces.

Looking at the Git history, I believe it was fixed in #18697. Since then, the tool updates `sdk.dir` in the `local.properties` file in the same way that it updates `flutter.sdk` and other settings.

---

For attempting to reproduce the issue, I conveniently have a broken Android SDK lying around in /usr/lib/android-sdk/ (in addition to a working one in my homedir):
```
$ export ANDROID_HOME=/usr/lib/android-sdk
$ flutter doctor -v
[✓] Flutter (Channel master, 3.8.0-16.0.pre.55, on Debian GNU/Linux 10 (buster) 4.19.0-23-amd64, locale en_US.UTF-8)
    • Flutter version 3.8.0-16.0.pre.55 on channel master at /home/greg/n/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0e2b0503f (14 hours ago), 2023-02-25 23:18:07 -0800
    • Engine revision 951e3febc1
    • Dart version 3.0.0 (build 3.0.0-272.0.dev)
    • DevTools version 2.22.1

[✗] Android toolchain - develop for Android devices
    • Android SDK at /usr/lib/android-sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
```

So, following the original repro steps: Step 2 fails as expected. And the broken Android SDK is referred to in the `local.properties` file:
```
$ grep ^sdk.dir= android/local.properties
sdk.dir=/usr/lib/android-sdk
```
But then Step 4 succeeds. And the `local.properties` has been updated:
```
$ grep ^sdk.dir= android/local.properties
sdk.dir=/home/greg/Android/Sdk
```
